exhibitor_tls_artifacts/gen_artifacts.py

```diff
@@ -4,46 +4,74 @@
 
 from .gen_certificates import CertificateGenerator
 from .gen_stores import KeystoreGenerator
+from .validators import validate_dir_missing
 
 
 @click.command()
-@click.argument('sans', nargs=-1)
-@click.option('-d', '--dir', help='Directory to put artifacts in.',
-              default='./artifacts/')
-def app(sans, dir):
-    os.makedirs(dir)
+@click.argument('nodes', nargs=-1)
+@click.option(
+    '-d',
+    '--output-directory',
+    help='Directory to put artifacts in. This output_directory must not exist.',
+    default='./artifacts/',
+    callback=validate_dir_missing)
+def app(nodes, output_directory):
+    """
+    Generates Admin Router and Exhibitor TLS artifacts. NODES should consist
+    of a space seperated list of master ip addresses. See
+    https://docs.mesosphere.com/1.13/security/ent/tls-ssl/exhibitor-tls/
+    """
+    if not nodes:
+        raise click.BadArgumentUsage('No nodes have been provided.')
 
-    if len(sans) < 1:
-        sans = ['localhost', '127.0.0.1']
+    # Create artifact output_directory
+    os.makedirs(output_directory)
```

Comment:
So, this program is supposed to error out if `output_directory` exists. What do we think about about catching `FileExistsError` and doing `sys.exit('good error message')`?

Reply:
@jgehrcke That's what the callback does, I believe - https://github.com/mesosphere/exhibitor-tls-artifacts-gen/pull/3/files#diff-25b812e37b704e22491fe6474c0d3ed5R5.

Reply:
@adamtheturtle thanks (:

Reply:
This is handled using a click callback:
https://github.com/jr0d/exhibitor-tls-artifacts-gen/blob/jr-add-client-certs/exhibitor_tls_artifacts/validators.py

The result:

```
$ touch existing
$ exhibitor-tls-artifacts -d existing
Usage: exhibitor-tls-artifacts [OPTIONS] [NODES]...

Error: Invalid value for "-d" / "--output-directory": Artifacts directory cannot exist!
```

Reply:
@jgehrcke ^